suppression-backup-service/exporter/exporter.go

```diff
@@ -0,0 +1,223 @@
+package exporter
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+	suppression "github.com/rudderlabs/rudder-server/enterprise/suppress-user"
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/suppression-backup-service/model"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+// ExporterOpt represents a configuration option for the exporter
+type ExporterOpt func(*Exporter)
+
+func Export(repo suppression.Repository, file model.File) error {
+	// export initially to a temp file
+	tempFile, err := os.CreateTemp("", "tmp-export")
+	if err != nil {
+		return fmt.Errorf("error creating temp file: %w", err)
+	}
+	defer func() {
+		_ = tempFile.Close()
+		_ = os.Remove(tempFile.Name())
+	}()
+
+	// export to temp file
+	err = repo.Backup(tempFile)
+	if err != nil {
+		return fmt.Errorf("error exporting suppression list: %w", err)
+	}
+
+	// move temp file to final destination
+	file.Mu.Lock()
+	defer file.Mu.Unlock()
+	err = os.Rename(tempFile.Name(), file.Path)
+	if err != nil {
+		return fmt.Errorf("error moving temp file to final destination: %w", err)
+	}
+
+	return nil
+}
+
+func newBadgerDBInstance(baseDir string, pkgLogger logger.Logger) (suppression.Repository, error) {
+	_ = os.RemoveAll(baseDir)
```

Comment:
1. In case of a full export, the badgerDB folders should be preserved between restarts, they shouldn't be deleted.
2. If `RemoveAll` fails, we shouldn't ignore this error.
3. Maybe this operation shouldn't be performed in this function at all, but just before it, wherever necessary.

Reply:
good point... modified.